wallet-new/server/Main.hs

```diff
@@ -1,211 +1,21 @@
-    plugins :: (PassiveWalletLayer IO, PassiveWallet)
-            -> Plugins.Plugin Kernel.Mode.WalletMode
-    plugins w = mconcat [ Plugins.walletBackend pm params w ]
-
-    -- Extract the logger name from node parameters
-    --
-    -- TODO: Not sure what the policy is for logger names of components.
-    -- For now we just use the one from the node itself.
-    logMessage' :: Severity -> Text -> IO ()
-    logMessage' sev txt =
-        usingLoggerName loggerName $ logMessage sev txt
-      where
-        loggerName :: LoggerName
-        loggerName = lpDefaultName . bpLoggingParams . npBaseParams $ nodeParams
-
--- | Runs an edge node plus its wallet backend API.
-startEdgeNode :: HasCompileInfo
-              => WalletStartupOptions
-              -> IO ()
-startEdgeNode wso =
-  withConfigurations blPath conf $ \ntpConfig pm -> do
-      (sscParams, nodeParams) <- getParameters ntpConfig
-      case wsoWalletBackendParams wso of
-        WalletLegacy legacyParams ->
-          actionWithWallet pm sscParams nodeParams ntpConfig legacyParams
-        WalletNew newParams ->
-          actionWithNewWallet pm sscParams nodeParams newParams
-  where
-    getParameters :: HasConfigurations => NtpConfiguration -> IO (SscParams, NodeParams)
-    getParameters ntpConfig = do
-
-      currentParams <- CLI.getNodeParams defaultLoggerName (wsoNodeArgs wso) nodeArgs
-      let vssSK = fromJust $ npUserSecret currentParams ^. usVss
-      let gtParams = CLI.gtSscParams (wsoNodeArgs wso) vssSK (npBehaviorConfig currentParams)
-
-      CLI.printInfoOnStart (wsoNodeArgs wso) ntpConfig
-      logInfo "Wallet is enabled!"
-
-      return (gtParams, currentParams)
-
-    conf :: ConfigurationOptions
-    conf = CLI.configurationOptions $ CLI.commonArgs (wsoNodeArgs wso)
-
-    blPath :: Maybe AssetLockPath
-    blPath = AssetLockPath <$> CLI.cnaAssetLockPath (wsoNodeArgs wso)
-
-    nodeArgs :: CLI.NodeArgs
-    nodeArgs = CLI.NodeArgs { CLI.behaviorConfigPath = Nothing }
-
+import           Cardano.Wallet.Launcher (startWalletNode)
+import           Cardano.Wallet.Server.CLI (WalletStartupOptions (..),
+                     getWalletNodeOptions)
+import           Pos.Client.CLI.NodeOptions (NodeArgs (..))
+import           Pos.Client.CLI.Params (loggingParams)
+import           Pos.Util.CompileInfo (withCompileInfo)
 
 -- | The main entrypoint for the Wallet.
 main :: IO ()
 main = withCompileInfo $ do
-    cfg <- getWalletNodeOptions
-    putText "Wallet is starting..."
-    let loggingParams = CLI.loggingParams defaultLoggerName (wsoNodeArgs cfg)
-    loggerBracket loggingParams . logException "node" $ do
-        logInfo "[Attention] Software is built with the wallet backend"
-        startEdgeNode cfg
+    putText "Wallet is starting...\n"
+    wOpts <- getWalletNodeOptions
+    let nArgs = NodeArgs { behaviorConfigPath = Nothing }
+    let lArgs = loggingParams "node" (wsoNodeArgs wOpts)
+    startWalletNode nArgs wOpts lArgs
```

Comment:
Here we leverage the `startWalletNode` that is now exposed in `Cardano/Wallet/Launcher`. Arguments comes from the CLI.